src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Core.cs

```diff
@@ -62,5 +64,69 @@ private void CheckDisposedOrStarted()
                 throw new InvalidOperationException(SR.net_http_operation_started);
             }
         }
+
+        private static async Task<Exception> ValidateAndNormalizeRequestAsync(HttpRequestMessage request)
+        {
+            bool shouldBufferContent = false;
+
+            // Add headers to define content transfer, if not present
+            if (request.HasHeaders && request.Headers.TransferEncodingChunked.GetValueOrDefault())
+            {
+                if (request.Content == null)
+                {
+                    return new HttpRequestException(SR.net_http_client_execution_error,
+                        new InvalidOperationException(SR.net_http_chunked_not_allowed_with_empty_content));
+                }
+
+                // Since the user explicitly set TransferEncodingChunked to true, we need to remove
+                // the Content-Length header if present, as sending both is invalid.
```

Comment:
I was recenty looking for code that does this; so nice :)

I couldn't get it output Transfer-Encoding and Content-Length similuatiously though; but couldn't find where it happened.

Reply:
FWIW it is a protocol violation to send both Transfer-Encoding and Content-Length headers - see https://tools.ietf.org/html/rfc7230#section-3.3.2

> A sender MUST NOT send a Content-Length header field in any message that contains a Transfer-Encoding header field.

This PR will allow you to opt out of chunked encoding though, so that Content-Length is calculated and  specified instead.

Reply:
Yeah, you can do request smuggling via proxies if they allow it (which they shouldn't) https://portswigger.net/blog/http-desync-attacks-request-smuggling-reborn

Reply:
Just making sure you were aware. There are duplicate checks in the WinHttpHandler, SocketsHttpHandler, and CurlHandler files. 

Search for “TransferEncodingChunked” in those class’s files and it’ll get you close to where those checks are implemented.